### PR TITLE
DDP-5758 - remove unused methods and remove unit tests testing it

### PR DESCRIPTION
**Pull request by evgeniipr**
## Context

Remove some methods from `QuestionDao.java` which unused anywhere except unit tests. Remove these unit tests either (because it tests the results of these removed methods).
_**I have doublts**: may be these methods should be preserved in order to preserve unit tests where it used? Because unit tests besides the direct test purposes - an additional documentation for the tested code. And actually it tests the Jdbi-methods which still not removed because used in some parts of code. @yufengwng what do you think?_

## Checklist

- [ ] I have labeled the type of changes involved using the `C-*` labels.
- [ ] I have assessed potential risks and labeled using the `R-*` labels.
- [ ] I have considered error handling and alerts, and added `L-*` labels as needed.
- [ ] I have considered security and privacy, and added `I-*` labels as needed
- [ ] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.
- [ ] If applicable, I have discussed the analytics needs at both a platform and study level with Product and instrumented code accordingly.
- [ ] If applicable, my UI/UX changes have passed muster with Product/Design via an over-the-shoulder review, screenshots, etc.

_If unsure or need help with any of the above items, add the `help wanted` label. For items that starts with `If applicable`, if it is not applicable, check it off and add `n/a` in front._

## FUD Score

_Overall, how are you feeling about these changes?_

- [ ] :relaxed: All good, business as usual!
- [ ] :sweat_smile: There might be some issues here
- [ ] :scream: I'm sweaty and nervous

## How do we demo these changes?

_How does one observe these changes in a deployed system? Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [ ] They are user-visible in dev as a regular user journey and require no additional instructions.
- [ ] Getting dev into a state where this is user-visible requires some tech fiddling. I have documented these steps in the related ticket.
- [ ] Requires other features before it's human visible. I have documented the blocking issues in jira.
- [ ] I have no idea how to demo this. Please help me!

## Testing

- [ ] I have written automated positive tests
- [ ] I have written automated negative tests
- [ ] I have written zero automated tests but have poked around locally to verify proper functionality
- [ ] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release

- [ ] These changes require no special release procedures--just code!
- [ ] Releasing these changes requires special handling and I have documented the special procedures in the release plan document

### CODE REVIEW

**Comment by ssettipalli** [comment]
We need to watch if these methods are meant to be used by unit tests whose main purpose is to test some  other functionality and not test these methods.

File: pepper-apis/src/test/java/org/broadinstitute/ddp/db/dao/QuestionDaoTest.java [hunk 1]
```diff
@@ -729,95 +725,6 @@ private GetGenericQuestionData genericGetTextQuestionSetUp(Handle handle) {
                 activityInstanceDto.getGuid(), activityInstanceDto.getCreatedAtMillis(), question, answers);
     }
 
-    @Test
-    public void testGetQuestionByBlockId_success() {
```
Comment (yufengwng) [inline pepper-apis/src/test/java/org/broadinstitute/ddp/db/dao/QuestionDaoTest.java:733]: These unit tests literally just tests a method that we no longer use. So result of the tests has no use to us. I recommend remove these unit tests.

File: pepper-apis/src/test/java/org/broadinstitute/ddp/db/dao/QuestionDaoTest.java [hunk 2]
```diff
@@ -1133,60 +952,6 @@ public void testGetPicklistQuestion_success() {
         });
     }
 
-    @Test
-    public void testGetPicklistQuestion_withGroups() {
```
Comment (yufengwng) [inline pepper-apis/src/test/java/org/broadinstitute/ddp/db/dao/QuestionDaoTest.java:1137]: For this and the "text question" tests below, do we have equivalent tests elsewhere? Looks like they're not just testing the removed methods. They're creating picklist/text questions and making we can fetch them. Have we sufficiently covered these in other tests?

Comment (evgeniipr) [reply pepper-apis/src/test/java/org/broadinstitute/ddp/db/dao/QuestionDaoTest.java:1137]: > For this and the "text question" tests below, do we have equivalent tests elsewhere? Looks like they're not just testing the removed methods. They're creating picklist/text questions and making we can fetch them. Have we sufficiently covered these in other tests?

So it needs to analyze more deeply the purpose of tests which use these methods.. 
I will try to do it.

Comment (yufengwng) [reply pepper-apis/src/test/java/org/broadinstitute/ddp/db/dao/QuestionDaoTest.java:1137]: I mean, I don't see any more "deep purpose" with these tests. They just create a question, insert it, and make sure we can read it back out. I'm sure we should have other tests that does this already.

**Review by Trimax — APPROVED** [review]

**Comment by bskinner** [comment]
@Trimax This PR is extremely out of date and should not be merged in as-is. Why are you approving it?

**Pull request closed without merging**